### ISSUE TITLE
[Regression] Auto indentation fails for multiple lines

### ISSUE THREAD

**Issue by adamreisnz** [atom/atom#16571]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description
![sigh](https://user-images.githubusercontent.com/490562/34974740-7444f40c-faf3-11e7-844e-63bc8fb1d69a.gif)

### Steps to Reproduce

1. Type code as in the example, e.g.

```js
    //Create change(s)
    for (const item of items) {
      await mongoose
        .model('Change')
        .create({type, item, timestamp, isDeleted});
    }```
2. Move the 3 lines with the `await` statement outside of the for loop.
3. Observe that auto indentation breaks the indentation.
4. Try the same when moving the code into the for loop.

**Expected behavior:** Correct indentation

**Actual behavior:** Incorrect indentation

**Reproduces how often:** 100%

### Versions

Atom    : 1.25.0-dev-390ab7449
Electron: 1.7.10
Chrome  : 58.0.3029.110
Node    : 7.9.0

apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.25.0-dev-390ab7449
python 2.7.10
git 2.14.1

OS X 10.11.5

### Additional information
See also https://github.com/atom/atom/issues/16556

This is a regression that only started happening in the last month or so in Atom's dev builds.
Are there no tests in place around these kind of indentation actions?

**Comment by rsese** [comment]
Thanks @adamreisnz - just to clarify, after this step:

> Move the 3 lines with the await statement outside of the for loop.

What does the indentation look like for you?  Just dropping what I see here (please let me know if I'm doing something wrong):

![auto-indent](https://user-images.githubusercontent.com/734194/35013959-a49b165a-fad4-11e7-9cf9-56151c2e11c3.gif)

**Comment by adamreisnz** [comment]
Thanks for looking into it. When I cut and paste, the indentation remains correct. The problem arises when the lines are moved up or down, using the Move line up or Move line down commands. I'll update the issue title and description accordingly.

**Comment by rsese** [comment]
Ahhh thanks, I can reproduce with the `Editor: Move Line Up/Down` commands - do you think this issue is already captured in https://github.com/atom/atom/issues/10524 or it's slightly different?

**Comment by adamreisnz** [comment]
No, I think it's different, because I wasn't suffering from this particular issue until only last week, when I built a new version of Atom again. My last build was from end of Dec, and that issue has been open since January 2016.

That issue also seems to be fairly generic and I don't really see it describing what the incorrect behaviour is. This issue on the other hand is quite specific.

**Comment by iolsen** [comment]
@ungb is going to confirm this is a regression.

**Comment by ungb** [comment]
I'm able to repro.  I updated the repro steps to be more clear and added this to our backlog.

**Comment by corysimmons** [comment]
I think it's the same problem but this issue is much nicer. I'd point other issue to this one and close it.

Is there a label for putting more priority on fixing this? I've just ceased moving blocks of text in Atom for years now and the other issue(s) have been open for years with lots of misc issues and thumbs. Seems like _a lot_ of people are bothered by this.

**Comment by sosaadat** [comment]
Has anyone learned any additional information on what is causing this issue?

**Comment by lee-dohm** [comment]
Does this still reproduce after the tree-sitter change, in other words in v1.32.0 or later?

**Comment by adamreisnz** [comment]
Still happening in 1.34 nightly (along with [several](https://github.com/atom/language-javascript/issues/597) newly introduced [bugs](https://github.com/atom/language-javascript/issues?q=is%3Aissue+is%3Aopen+label%3Atree-sitter) related to Tree Sifter)
